### DISCUSSION TITLE
problem when trying to push repository

### DISCUSSION THREAD

**Discussion by creapanda** [community/community#121280] [New to GitHub] [answered]
Hi, I'm new to GitHub and I have encountered this problem for days now. the problem is I can't push my repository because "write access to repository not granted." I work in a group and we share the same repository, this only happens on my laptop. Does anyone know what is the source of this problem?

**Comment by niladri-1** [comment]
## Hi, @creapanda, this will solve the problem

1. **Check Repository Permissions**: Confirm that you have been granted write access to the repository. You can do this by visiting the repository on GitHub and looking at the "Settings" tab, then selecting "Manage access" to see who has access and at what level.

2. **Local Git Configuration**: Make sure that your local Git configuration is set up correctly. Check that you're using the correct username and email address associated with your GitHub account by running the following commands in your terminal:

   ```
   git config --global user.name "Your Name"
   git config --global user.email "your.email@example.com"
   ```

3. **Credentials**: Ensure that your GitHub credentials are correctly configured on your laptop. You may need to re-authenticate with GitHub if your access token or password has changed recently.

4. **HTTPS vs. SSH**: Check if you're using HTTPS or SSH for your Git remote. If you're using HTTPS, you may need to authenticate with your GitHub username and password. If you're using SSH, make sure your SSH key is correctly set up and added to your GitHub account.

5. **Push Command**: When pushing changes to the repository, make sure you're using the correct command:

   ```
   git push origin <branch_name>
   ```

   Replace `<branch_name>` with the name of the branch you want to push your changes to.

6. **Errors and Logs**: If you encounter any specific error messages when trying to push, pay attention to them as they can provide clues about what might be going wrong. Check the Git output and any error messages for more information.

7. **Reach Out for Help**: If you've tried all of the above steps and are still encountering issues, consider reaching out to the repository owner or an administrator for further assistance. They may be able to provide insights or adjust permissions as needed.

By following these steps and troubleshooting each aspect carefully, you should be able to identify and resolve the issue with pushing to the repository from your laptop.

**Comment by creapanda** [comment]
thank you, I have tried everything except number 1 cause I can not find the "setting" tag, so I'm trying to contact the repository owner

**Reply by gurjeetsinghvirdee** [reply, marked as answer]
As you are not the repository owner, that's why the `Settings` tab is not visible to you. Ask the repository owner to send you a collaboration invitation, then you can push changes directly without forking.
